File: app/src/main/java/com/example/blooddonationapp/Fragments/MyRequestsFragment.java
package com.example.blooddonationapp.Fragments;

import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.blooddonationapp.Adapters.MyRequestAdapter;
import com.example.blooddonationapp.FirebaseAuthSingleton;
import com.example.blooddonationapp.Models.MyRequestItem;
import com.example.blooddonationapp.R;
import com.firebase.ui.firestore.FirestoreRecyclerAdapter;
import com.firebase.ui.firestore.FirestoreRecyclerOptions;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.FirebaseFirestoreException;
import com.google.firebase.firestore.Query;

import java.util.Objects;

public class MyRequestsFragment extends Fragment {
    private FirebaseAuthSingleton mAuth = FirebaseAuthSingleton.INSTANCE;
    private FirebaseFirestore db;
    RecyclerView request_item;
    private FirestoreRecyclerAdapter request_adapter;
    LinearLayoutManager requestLayoutManager;


    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {

        View root = inflater.inflate(R.layout.my_requests_fragment, container, false);


        request_item = root.findViewById(R.id.list);
        init(root.getContext());


        getRequestsList(root.getContext());


        return root;
    }

    private void init(Context root) {
        requestLayoutManager = new LinearLayoutManager(root.getApplicationContext(), LinearLayoutManager.VERTICAL, false);
        request_item.setLayoutManager(requestLayoutManager);
        db = FirebaseFirestore.getInstance();
    }

    private void getRequestsList(Context root) {
        Query query = db.collection("requests").whereEqualTo("user_id", mAuth.getInstance().getCurrentUser().getUid().toString());

        FirestoreRecyclerOptions<MyRequestItem> response = new FirestoreRecyclerOptions.Builder<MyRequestItem>()
                .setQuery(query, MyRequestItem.class)
                .build();

        request_adapter = new MyRequestAdapter(response);
        request_item.setAdapter(request_adapter);
    }


    @Override
    public void onStart() {
        super.onStart();
        request_adapter.startListening();

    }
}

File: app/src/main/java/com/example/blooddonationapp/fragments/MyRequestsFragment.kt
package com.example.blooddonationapp.fragments

import android.content.Context
import com.example.blooddonationapp.utils.FirebaseAuthSingleton
import com.google.firebase.firestore.FirebaseFirestore
import androidx.recyclerview.widget.RecyclerView
import com.firebase.ui.firestore.FirestoreRecyclerAdapter
import androidx.recyclerview.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import com.example.blooddonationapp.R
import com.firebase.ui.firestore.FirestoreRecyclerOptions
import com.example.blooddonationapp.models.MyRequestItem
import com.example.blooddonationapp.adapters.MyRequestAdapter

class MyRequestsFragment : Fragment() {
    private val mAuth = FirebaseAuthSingleton
    private var db: FirebaseFirestore? = null
    var request_item: RecyclerView? = null
    private var request_adapter: FirestoreRecyclerAdapter<*, *>? = null
    var requestLayoutManager: LinearLayoutManager? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View? {
        val root = inflater.inflate(R.layout.my_requests_fragment, container, false)
        request_item = root.findViewById(R.id.list)
        init(root.context)
        getRequestsList()
        return root
    }

    private fun init(root: Context) {
        requestLayoutManager =
            LinearLayoutManager(root.applicationContext, LinearLayoutManager.VERTICAL, false)
        request_item!!.layoutManager = requestLayoutManager
        db = FirebaseFirestore.getInstance()
    }

    private fun getRequestsList() {
        val query = db!!.collection("requests").whereEqualTo(
            "user_id", mAuth.instance!!.currentUser!!
                .uid
        )
        val response = FirestoreRecyclerOptions.Builder<MyRequestItem>()
            .setQuery(query, MyRequestItem::class.java)
            .build()
        request_adapter = MyRequestAdapter(response)
        request_item!!.adapter = request_adapter
    }

    override fun onStart() {
        super.onStart()
        request_adapter!!.startListening()
    }
}